nautobot_secrets_providers/providers/hashicorp.py

```diff
@@ -41,19 +49,101 @@ class ParametersForm(BootstrapMixin, forms.Form):
+        elif auth_method == "token":
+            if "token" not in vault_settings:
+                raise exceptions.SecretProviderError(
+                    secret, cls, "HashiCorp Vault configuration is missing a token for token authentication!"
+                )
+        elif auth_method == "kubernetes":
+            if "role_name" not in vault_settings:
+                raise exceptions.SecretProviderError(
+                    secret, cls, "HashiCorp Vault configuration is missing a role name for kubernetes authentication!"
+                )
+        elif auth_method == "approle":
+            if "role_id" not in vault_settings or "secret_id" not in vault_settings:
+                raise exceptions.SecretProviderError(
+                    secret, cls, "HashiCorp Vault configuration is missing a role_id and/or secret_id!"
+                )
+
+        return vault_settings
+
+    @classmethod
+    def get_client(cls, secret=None):
+        """Authenticate and return a hashicorp client."""
+        vault_settings = cls.validate_vault_settings(secret)
+        auth_method = vault_settings.get("auth_method", "token")
+
+        k8s_token_path = vault_settings.get("k8s_token_path", K8S_TOKEN_DEFAULT_PATH)
+        login_kwargs = vault_settings.get("login_kwargs", {})
+
+        # According to the docs (https://hvac.readthedocs.io/en/stable/source/hvac_v1.html?highlight=verify#hvac.v1.Client.__init__)
+        # the client verify parameter is either a boolean or a path to a ca certificate file to verify.  This is non-intuitive
+        # so we use a parameter to specify the path to the ca_cert, if not provided we use the default of None
+        ca_cert = vault_settings.get("ca_cert", None)
+
+        # Get the client and attempt to retrieve the secret.
+        try:
+            if auth_method == "token":
+                client = hvac.Client(url=vault_settings["url"], token=vault_settings["token"], verify=ca_cert)
+            else:
+                client = hvac.Client(url=vault_settings["url"], verify=ca_cert)
+                if auth_method == "approle":
+                    client.auth.approle.login(
+                        role_id=vault_settings["role_id"],
+                        secret_id=vault_settings["secret_id"],
+                        **login_kwargs,
+                    )
+                elif auth_method == "kubernetes":
+                    with open(k8s_token_path, "r", encoding="utf-8") as token_file:
+                        jwt = token_file.read()
+                    client.auth.kubernetes.login(role=vault_settings["role_name"], jwt=jwt, **login_kwargs)
+                elif auth_method == "aws":
+                    session = boto3.Session()
+                    aws_creds = session.get_credentials()
+                    client.auth.aws.iam_login(
+                        access_key=aws_creds.access_key,
+                        secret_key=aws_creds.secret_key,
+                        session_token=aws_creds.token,
```

Comment:
Might be good to explicitly document in README that the AWS credentials *cannot* be specified via `vault_settings`, they must be pre-configured for AWS directly (and link to the appropriate docs)?

Alternately, should we for consistency with other methods at least provide an *option* to set the AWS credentials via `vault_settings`?

Reply:
from a security perspective, and from and AWS best practices stance, I'll just say not just no but absolutely not!  We handle the AWS authentication the same way as AWS secrets manager, just to connect to Vault instead.  I'll update the docs.